modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -446,7 +446,8 @@ protected void configureDefaultHttpPipeline(ChannelPipeline pipeline) {
             );
             decoder.setCumulator(ByteToMessageDecoder.COMPOSITE_CUMULATOR);
             pipeline.addLast("decoder", decoder);
-            pipeline.addLast("decoder_compress", new HttpContentDecompressor());
+            pipeline.addLast("header_verifier", transport.createHeaderVerifier());
```

Comment:
If `header_verifier` is noop (not configured), we should not include it into the pipeline

Reply:
Does it make sense to have another overridable extension point?

In addition to createHeaderVerifier, should there be a `hasHeaderVerifier` as well that configures this step in the pipeline if true?

Reply:
If API return `Optional<HttpHeaderVerifier>` - you will get it figured out

Reply:
I think its easier to read that the element is always inserted and it is a noop vs adding branching logic to include or not this item.  I'm a fan of noop implementations and I think this is a good case for one.

Reply:
>  I'm a fan of noop implementations and I think this is a good case for one.
This is not a noop in that sense - we adding more handlers to Netty pipeline that, well, have to be traversed and executed, even if doing nothing.